assets/src/data/persistence/objectives.ts

```diff
@@ -0,0 +1,77 @@
+import { ObjectiveMap } from '../content/activity';
+
+/**
+ * In the backend data model, objectives are attached to parts.
+ * But in adaptive lessons, we want to attach them to an activity, so
+ * we're going to "cheat" and attach them to the first part in the activity.
+ * However there's a few use cases we need to handle:
+ *
+ * 1. If there are no parts, a __default part is created, so attach it to that.
+ * 2. If the part that objectives are attached to is deleted, we should move the objectives
+ *    to another part.
+ * 3. If there were no parts, and objectives were attached, and later a part is added, that
+ *    default part is deleted, so move the objectives to the newly created part.
+ *
```

Comment:
just curious about the reasoning behind not using a dedicated `__objectives` part as opposed to managing the part shifts

Reply:
It didn't seem harder to do it this way, and results in fewer extra nodes in the data. I'm assuming for very close to 100% of actual real-life usage, we don't need __default or __objectives in a real scorable activity.

I also had some worry about analytics not picking up the part some day, but in the end, that was probably unfounded.